script.js

```diff
@@ -0,0 +1,105 @@
+        lettersOfSelection.unshift("**")
+        lettersOfSelection.push("**")
+        let solidString = lettersOfSelection.join('')
+        sourceTextBox.setRangeText(solidString)
+        return
+    }
+    else if (lettersOfSelection[0] === ("*") && lettersOfSelection[1] === ("*")) {
+        lettersOfSelection.shift()
+        lettersOfSelection.shift()
+        lettersOfSelection.pop()
+        lettersOfSelection.pop()
+        let solidString = lettersOfSelection.join('')
+        sourceTextBox.setRangeText(solidString)
+        return
+    }
+    else if (lettersOfSelection[0] !== ("*") && lettersOfSelection[1] !== ("*")) {
+        lettersOfSelection.unshift("**")
+        lettersOfSelection.push("**")
+        let solidString = lettersOfSelection.join('')
+        sourceTextBox.setRangeText(solidString)
+        return
+    }
+    else {
+        return
+    }
+}
+
+
+function italText() {
+    let lettersOfSelection = selection.split('')
+    if (selection === ("")) {
+        return
+    }
+    else if (lettersOfSelection[0] !== ("*") && lettersOfSelection[lettersOfSelection.length - 1] === ("*")) {
+        return
+    }
+    else if (lettersOfSelection[0] !== ("*") && lettersOfSelection[lettersOfSelection.length - 1] !== ("*")) {
+        lettersOfSelection.unshift("*")
+        lettersOfSelection.push("*")
+        let solidString = lettersOfSelection.join('')
+        sourceTextBox.setRangeText(solidString)
+        return
+    }
+    else if (lettersOfSelection[0] === ("*") && lettersOfSelection[1] !== ("*") && lettersOfSelection[lettersOfSelection.length - 1] === ("*") && lettersOfSelection[lettersOfSelection.length - 2] !== ("*")) {
+        lettersOfSelection.shift()
+        lettersOfSelection.pop()
+        let solidString = lettersOfSelection.join('')
+        sourceTextBox.setRangeText(solidString)
+        return
+    }
+    else if (lettersOfSelection[0] === ("*") && lettersOfSelection[1] === ("*") && lettersOfSelection[2] !== ("*") && lettersOfSelection[lettersOfSelection.length - 2] === ("*") && lettersOfSelection[lettersOfSelection.length - 1] === ("*")) {
+        lettersOfSelection.unshift("*")
+        lettersOfSelection.push("*")
+        let solidString = lettersOfSelection.join('')
+        sourceTextBox.setRangeText(solidString)
+        return
+    }
+    else if (lettersOfSelection[0] === ("*") && lettersOfSelection[1] === ("*") && lettersOfSelection[2] === ("*")) {
+        lettersOfSelection.shift()
+        lettersOfSelection.pop()
+        let solidString = lettersOfSelection.join('')
+        sourceTextBox.setRangeText(solidString)
+        return
+    }
+    else {
+        return
+    }
+}
+
+
+
+sourceTextBox.placeholder = ("Type Commands in this box...")
+sourceTextBox.addEventListener('input', markdownText)
+sourceTextBox.addEventListener('select', selectedText)
```

Comment:
I see where I was confused earlier. `selectedText` sounds to me like a function that returns the selected text; instead it's an event handler that grabs the selected text *and* puts it into a global var. Renaming it to `selectText` or `grabSelectedText` or something more active would have helped me.